### PR TITLE
Background sky and landscapes

### PR DESCRIPTION
**Pull request by rtr-dnd**
## Context

Gradient sky, and gradient+stripe landscapes.
This has to be merged after #6 got merged, because of name conflicts.

## Summary

Created gradient sky and gradient+stripe landscapes.

## How to check behavior

Clone and start zennist.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (Aki-7) [inline package.json]: Are `package.json` and `package-locak.json` necessary?

File: shaders/floor.vert [hunk 2]
```diff
@@ -2,9 +2,13 @@
...
+
+out vec2 uv;
 
 void
 main()
 {
   gl_Position = zMVP * position;
-}
+  uv = uv_in;
+}
```
Comment (Aki-7) [inline shaders/floor.vert]: Adding newline is better.

File: shaders/landscape.frag [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#version 320 es
+#define M_PI 3.1415
+precision mediump float;
+
+uniform vec4 color_base1;
+uniform vec4 color_base2;
+uniform vec4 color_stripe;
+// uniform float freq;
+// uniform float threshold;
```
Comment (Aki-7) [inline shaders/landscape.frag]: Isn't it impossible to use these variables instead of `in_vec`?

File: shaders/sky.frag [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+in vec4 pos;
+in vec2 uv;
+out vec4 out_color;
+
+void
+main()
+{
+  out_color = mix(color1, color2, uv.y * 2.0);
+}
```
Comment (Aki-7) [inline shaders/sky.frag]: Please add newline.
My recommendation is to configure your editor to automatically insert newline at the end of file.

File: src/floor.cc [hunk 5]
```diff
@@ -112,51 +113,15 @@ Floor::Init()
...
+    float theta = M_PI * 2.f * i / float(resolution);
+    float x = radius_ * cosf(theta);
+    float y = 0.2f;
+    float z = radius_ * sinf(theta);
+    vertices_.emplace_back(x, y, z, x, z);
   }
 }
 
-}  // namespace zennist
+}  // namespace zennist
```
Comment (Aki-7) [inline src/floor.cc]: Here as well.

File: src/floor.h [hunk 6]
```diff
@@ -70,16 +63,4 @@ Floor::vertex_buffer_size()
...
-}
-
-inline size_t
-Floor::pool_size()
-{
-  return vertex_buffer_size() + element_array_buffer_size();
-}
-
-}  // namespace zennist
+}  // namespace zennist
```
Comment (Aki-7) [inline src/floor.h]: Here as well.

File: src/landscape.cc [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+        float x0 = mountains[k][0];
+        float z0 = mountains[k][1];
+        float h = mountains[k][2];
+        float flatness = mountains[k][3];
+        float r = sqrtf(powf(x - x0, 2) + powf(z - z0, 2));
+        if (r > flatness) continue;
+        y += h * (cosf(M_PI * r / flatness) + 1.f) / 2.f;
+      }
+
+      vertices_.emplace_back(x, y, z, u, v);  // (x1, y, z1), (x2, y, z1), ...
```
Comment (Aki-7) [inline src/landscape.cc]: What does this comment mean?

File: src/main.cc [hunk 8]
```diff
@@ -2,20 +2,64 @@
...
+#include "landscape.h"
+#include "jpeg-texture.h"
+#include "sphere.h"
 #include "floor.h"
 
 namespace zennist {
 
 class Application final : public zukou::IExpansiveDelegate
 {
+  glm::vec4 rgbColor(float r, float g, float b)
```
Comment (Aki-7) [inline src/main.cc]: According to our name convention,
```suggestion
  glm::vec4 RgbColor(float r, float g, float b)
```

And this function seems better to be placed outside of `Application` class

File: src/sphere.cc [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+
+#include <sys/mman.h>
+#include <unistd.h>
+
+#include <cstring>
+#include <iostream>
+
+#include "color.frag.h"
+#include "default.vert.h"
+// #include "texture.frag.h"
```
Comment (Aki-7) [inline src/sphere.cc]:
```suggestion
```

File: package.json (hunk 1 above)
Comment (rtr-dnd) [reply package.json]: no it's probably a mistake! will remove

File: shaders/landscape.frag (hunk 3 above)
Comment (rtr-dnd) [reply shaders/landscape.frag]: @Aki-7 
I tried but it seemed to me that `zukou::BaseTechnique::Uniform()` only accepts `glm::mat` arguments. Maybe I'm wrong.
I thought using `glm::vec<1, float>` would be a workaround, but there's no `vec1` in GLSL so I decided to use `vec2` instead.

File: shaders/sky.frag (hunk 4 above)
Comment (rtr-dnd) [reply shaders/sky.frag]: Thanks, I configured VSCode to do that!

File: src/landscape.cc (hunk 7 above)
Comment (rtr-dnd) [reply src/landscape.cc]: This was just a personal memo. will remove

File: src/main.cc (hunk 8 above)
Comment (rtr-dnd) [reply src/main.cc]: Thanks. Will move it outside.

File: shaders/landscape.frag (hunk 3 above)
Comment (Aki-7) [reply shaders/landscape.frag]: You can use `glm::vec<1, float>` for `float` in GLSL. Sorry for less documentation.

Comment (rtr-dnd) [reply shaders/landscape.frag]: Oh ok! Will try out!

**Pull request merged**